ocs_ci/resiliency/resiliency_helper.py

```diff
@@ -352,3 +361,231 @@ def run_failure_case(self):
+            timeout: Maximum time to wait in seconds for each thread
+
+        Returns:
+            bool: True if all threads completed, False if any timed out
+        """
+        all_completed = True
+        for thread in self._scaling_threads:
+            if not self.wait_for_scaling_completion(thread, timeout):
+                all_completed = False
+        return all_completed
+
+    def scale_workloads_synchronously(self, workloads):
+        """
+        Scale all workloads synchronously (blocking operation).
+
+        Args:
+            workloads: List of workload objects to scale
+        """
+        log.info(f"Starting synchronous scaling of {len(workloads)} workloads")
+
+        for workload in workloads:
+            if self.scale_single_workload(workload):
+                log.info(
+                    f"Successfully scaled workload {workload.workload_impl.deployment_name}"
+                )
+            else:
+                log.info(
+                    f"Workload {workload.workload_impl.deployment_name} already at limits, skipping"
+                )
+
+    def scale_workloads_parallel(self, workloads, max_workers=None):
+        """
+        Scale all workloads in parallel (blocking operation).
+
+        Args:
+            workloads: List of workload objects to scale
+            max_workers: Maximum number of worker threads. If None, uses len(workloads)
+        """
+        if max_workers is None:
+            max_workers = len(workloads)
+
+        log.info(
+            f"Starting parallel scaling of {len(workloads)} workloads with {max_workers} workers"
+        )
+
+        with ThreadPoolExecutor(max_workers=max_workers) as executor:
+            # Submit all scaling tasks
+            scaling_futures = {
+                executor.submit(self.scale_single_workload, workload): workload
+                for workload in workloads
+            }
+
+            # Wait for all scaling operations to complete
+            for future in as_completed(scaling_futures):
+                workload = scaling_futures[future]
+                try:
+                    future.result()  # This will raise exception if scaling failed
+                    log.info(
+                        f"Successfully scaled workload {workload.workload_impl.deployment_name}"
+                    )
+                except CancelledError:
+                    log.warning(
+                        f"Scaling task was cancelled for workload {workload.workload_impl.deployment_name}"
+                    )
+                except TimeoutError:
+                    log.error(
+                        f"Scaling task timed out for workload {workload.workload_impl.deployment_name}"
+                    )
+                except (ValueError, RuntimeError) as e:
+                    log.error(
+                        f"Error while scaling workload {workload.workload_impl.deployment_name}: {e}"
+                    )
+                except Exception as e:
```

Comment:
same, are you sure you want to catch any exception? it could mask un expected issues

Reply:
Removed it